pkg/services/ngalert/store/alert_rule.go

```diff
@@ -1033,12 +1034,26 @@ func (st DBstore) RenameTimeIntervalInNotificationSettings(
 	return result, nil, st.UpdateAlertRules(ctx, updates)
 }
 
-func ruleConstraintViolationToErr(rule ngmodels.AlertRule, err error) error {
+func ruleConstraintViolationToErr(sess *db.Session, rule ngmodels.AlertRule, err error, logger log.Logger) error {
 	msg := err.Error()
 	if strings.Contains(msg, "UQE_alert_rule_org_id_namespace_uid_title") || strings.Contains(msg, "alert_rule.org_id, alert_rule.namespace_uid, alert_rule.title") {
-		return ngmodels.ErrAlertRuleConflict(rule, ngmodels.ErrAlertRuleUniqueConstraintViolation)
+		// return verbose conflicting alert rule error response
+		// see: https://github.com/grafana/grafana/issues/89755
+		var fetched_uid string
+		var existingPartialAlertRule ngmodels.AlertRule
+		ok, uid_fetch_err := sess.Table("alert_rule").Cols("uid").Where("org_id = ? AND title = ? AND namespace_uid = ?", rule.OrgID, rule.Title, rule.NamespaceUID).Get(&fetched_uid)
+		if uid_fetch_err != nil {
+			logger.Error("Error fetching uid from alert_rule table", "reason", uid_fetch_err.Error())
+		}
+		if ok {
+			existingPartialAlertRule = ngmodels.AlertRule{UID: fetched_uid, Title: rule.Title, NamespaceUID: rule.NamespaceUID}
+		}
+		return ngmodels.ErrAlertRuleConflictVerbose(existingPartialAlertRule, rule, ngmodels.ErrAlertRuleUniqueConstraintViolation)
 	} else if strings.Contains(msg, "UQE_alert_rule_org_id_uid") || strings.Contains(msg, "alert_rule.org_id, alert_rule.uid") {
-		return ngmodels.ErrAlertRuleConflict(rule, errors.New("rule UID under the same organisation should be unique"))
+		// return verbose conflicting alert rule error response
+		// see: https://github.com/grafana/grafana/issues/89755
+		existingPartialAlertRule := ngmodels.AlertRule{UID: rule.UID, Title: rule.Title, NamespaceUID: rule.NamespaceUID}
```

Comment:
The existing rule is not guaranteed to have the same Title or Namespace, I would just leave it as:
```suggestion
		existingPartialAlertRule := ngmodels.AlertRule{UID: rule.UID}
```